### PR TITLE
Ddpb 1723 incorrect report visibility

### PR DESCRIPTION
**Pull request by jackgoodby**

### CODE REVIEW

File: src/AppBundle/Resources/views/Pa/ClientProfile/overview.html.twig [hunk 1]
```diff
@@ -316,7 +316,7 @@
     } %}
 
     {% include 'AppBundle:Pa/ClientProfile:_reports.html.twig' with {
-    reports: report.client.reports,
+    reports: report.client.getSubmittedReports(),
```
Comment (elvisciotti) [inline src/AppBundle/Resources/views/Pa/ClientProfile/overview.html.twig]: report.client.submittedReports would be more in line

Comment (jackgoodby) [reply src/AppBundle/Resources/views/Pa/ClientProfile/overview.html.twig]: sure

File: src/AppBundle/Entity/Client.php [hunk 2]
```diff
@@ -716,4 +716,18 @@ public function getCoDeputies()
         }
         return array_values($coDeps);
     }
-}
+
+    /**
+     * @return array $submittedReports an array of submitted reports
+     */
+    public function getSubmittedReports()
```
Comment (elvisciotti) [inline src/AppBundle/Entity/Client.php]: we get a list of submitted reports from lay_home page as well.
ReportController:84-91

What about some refactor to use this new method there?

a method getActiveReport  can be added alongside. And the the arsort can be left out in the controller

File: tests/behat/features/pa/03-report/01-102/01-report-edit-period.feature [hunk 3]
```diff
@@ -1,5 +1,11 @@
 Feature: PA report
 
+  Scenario: PA does not see unsubmitted reports in the submitted reports section
+    Given I load the application status from "team-users-complete"
+    And I am logged in as "behat-pa1@publicguardian.gsi.gov.uk" with password "Abcd1234"
+    When I click on "pa-report-open" in the "client-1000014" region
+    Then I should see "No submitted reports" in the "client-profile-reports" region
```
Comment (elvisciotti) [inline tests/behat/features/pa/03-report/01-102/01-report-edit-period.feature]: Is there already a test that assert that at least one submitted report is listed ?

File: src/AppBundle/Entity/Client.php (hunk 2 above)
Comment (jackgoodby) [reply src/AppBundle/Entity/Client.php]: Good call - Done!

File: src/AppBundle/Controller/Report/ReportController.php [hunk 4]
```diff
@@ -78,16 +78,9 @@ public function indexAction(Request $request)
...
-
-        $reportActive = null;
-        $reportsSubmitted = [];
-        foreach ($reports as $currentReport) {
-            if ($currentReport->getSubmitted()) {
-                $reportsSubmitted[] = $currentReport;
-            } else {
-                $reportActive = $currentReport;
-            }
+        $reportsSubmitted = $client ? $client->getSubmittedReports() : [];
```
Comment (elvisciotti) [inline src/AppBundle/Controller/Report/ReportController.php]: @jackgoodby the method doesn't do the arsort, so you are altering the logic here. You might need to re-add it on $reportsSubmitted (not inside the method, as it's not always wanted I think). Might be not neeed, but you need to check the usage to be sure

Comment (jackgoodby) [reply src/AppBundle/Controller/Report/ReportController.php]: arsort now line 82

**Pull request merged**